packages/slice-machine/lib/builders/SliceBuilder/FieldZones/index.tsx

```diff
@@ -183,32 +216,61 @@ const FieldZones: FC = () => {
-        dataTip={dataTipText2}
-        widgetsArray={sliceBuilderWidgetsArray}
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        fields={variation.items}
-        EditModal={EditModal}
-        onDeleteItem={_onDeleteItem(WidgetsArea.Items)}
-        onSave={_onSave(WidgetsArea.Items)}
-        onSaveNewField={_onSaveNewField(WidgetsArea.Items)}
-        onDragEnd={_onDragEnd(WidgetsArea.Items)}
-        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-        poolOfFieldsToCheck={variation.items || []}
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment, @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-member-access
-        renderHintBase={({ item }) => `item${transformKeyAccessor(item.key)}`}
-        renderFieldAccessor={(key) =>
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-          `slice.items[i]${transformKeyAccessor(key)}`
-        }
-        testId="slice-repeatable-zone"
-        isRepeatableCustomType={undefined}
-      />
+      {hasItems ? (
+        <Zone
+          zoneType="slice"
+          zoneTypeFormat={undefined}
+          tabId={undefined}
+          isRepeatable
+          title="Repeatable Zone"
+          dataTip={dataTipText2}
+          widgetsArray={sliceBuilderWidgetsArray}
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          fields={variation.items}
+          EditModal={EditModal}
+          onDeleteItem={_onDeleteItem(WidgetsArea.Items)}
+          onSave={_onSave(WidgetsArea.Items)}
+          onSaveNewField={_onSaveNewField(WidgetsArea.Items)}
+          onDragEnd={_onDragEnd(WidgetsArea.Items)}
+          // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+          poolOfFieldsToCheck={variation.items || []}
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment, @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-member-access
+          renderHintBase={({ item }) => `item${transformKeyAccessor(item.key)}`}
+          renderFieldAccessor={(key) =>
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+            `slice.items[i]${transformKeyAccessor(key)}`
+          }
+          testId="slice-repeatable-zone"
+          isRepeatableCustomType={undefined}
+        />
+      ) : null}
+      <Dialog
+        size="small"
+        open={isDeleteRepeatableZoneDialogOpen}
+        onOpenChange={(open) => setIsDeleteRepeatableZoneDialogOpen(open)}
+      >
+        <DialogHeader icon="delete" title="Delete field" />
+        <DialogContent>
+          <Box padding={24} gap={12} flexDirection="column">
+            <strong>
+              This action will permanently remove the Repeatable Zone from the{" "}
+              {slice.model.name} slice.
+            </strong>
+            <div>
+              If you need repeatable fields again, use a group field in place of
```

Comment:
_✍️ potential change_

`If you need repeatable fields again, use a repeatable group field in place of the repeatable zone.`

Reply:
Good catch!